[nnc] Simplify i%100 to i if i is less than 100; fixed #52580

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62138
* __->__ #60693
* #60688

Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

huiguoo committed:
[nnc] Simplify i%100 to i if i is less than 100; fixed #52580

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

ZolotukhinM commented on test/cpp/tensorexpr/test_simplify.cpp:
```diff
@@ -1291,6 +1291,30 @@ TEST(Simplify, SimplifyDivWithLoopContext2) {
+  VarHandle i("i", kInt);
+  BufHandle a_buf("A", {100}, kInt);
+  auto forstmt = For::make(
+      i, 0, 100, Store::make(a_buf, {i}, (i % ExprHandle(100))));
```
```suggestion
      i, 0, 100, Store::make(a_buf, {i}, (i % 100)));
```

ZolotukhinM commented on test/cpp/tensorexpr/test_simplify.cpp:
```diff
@@ -1291,6 +1291,30 @@ TEST(Simplify, SimplifyDivWithLoopContext2) {
+  auto forstmt = For::make(
+      i, 0, 100, Store::make(a_buf, {i}, (i % ExprHandle(100))));
+
+  const Stmt* simplified = IRSimplifier::simplify(forstmt);
```
Nit: for_stmt

ZolotukhinM reviewed (approved):
LGTM!

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).

huiguoo committed:
Update on "[nnc] Simplify i%100 to i if i is less than 100; fixed #52580"


Differential Revision: [D29375938](https://our.internmc.facebook.com/intern/diff/D29375938)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29375938).
